### PR TITLE
Split the posts and magpie module apart

### LINKED ISSUE #20: Improving the getter methods inside the Keeper
Currently the `GetX` methods inside the `keeper.go` file have a single return value. As examples we can take a look at 

https://github.com/desmos-labs/desmos/blob/18c63f5e7f4b61ac43b705c70a789a12b2dee698/x/magpie/keeper.go#L41

https://github.com/desmos-labs/desmos/blob/18c63f5e7f4b61ac43b705c70a789a12b2dee698/x/magpie/keeper.go#L102

I personally think that in this case having a single value may make the "not found" situation hard to handle. 

Let's consider the first case (`GetPost`). As this returns a default `Post` when none is found, what should we check to make sure that is was not found indeed? 

I think that having a double return value such as the following might be more useful: 

```golang
func (k Keeper) GetPost(ctx sdk.Context, id string) (post Post, found bool) {
    // ...
} 
```

Having a method like this one, we could then perform a simple check like this one 

```golang 
if post, found := keeper.GetPost(ctx, id); found {
  // do something with post 
}
```

Or something like this 

```golang 
post, found := keeper.GetPost(ctx, id) 
if !found {
  // throw an error 
}

// do something with post
```

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description
Split the `magpie` module into `magpie` and `posts` (closes #20)

## Checklist
- [x] Targeted PR against correct branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: x/magpie/client/cli/tx.go [hunk 1]
```diff
@@ -19,124 +19,24 @@ import (
...
-		},
-	}
-}
-
 // GetCmdCreateSession is the CLI command for creating a session for create post
 func GetCmdCreateSession(cdc *codec.Codec) *cobra.Command {
 	return &cobra.Command{
 		Use:   "create-session [namespace] [external address] [pubkey] [external signer signature]",
-		Short: "record a session for external service to post a magpie",
+		Short: "Creates a session for an external service to post a magpie",
```
Comment (kwunyeung) [inline x/magpie/client/cli/tx.go]: Should not read "post a magpie" here as now `Magpie` will just be the bridge. Maybe just `Creates a session for an external service to post`

File: x/magpie/internal/keeper/handler.go [hunk 2]
```diff
@@ -162,33 +38,50 @@ func handleMsgCreateSession(ctx sdk.Context, keeper Keeper, msg types.MsgCreateS
...
 	copy(pkBytes33[:], pkBytes)
 	pubkey := secp256k1.PubKeySecp256k1(pkBytes33)
 
-	message := fmt.Sprintf(`{"account_number":"0","chain_id":"%s","fee":{"amount":[],"gas":"200000"},"memo":"","msgs":[{"type":"desmos/MsgCreateSession","value":{"created":"%s","external_owner":"%s","namespace":"%s","owner":"%s","pubkey":"%s","signature":null}}],"sequence":"0"}`,
-		ctx.ChainID(), msg.Created.Format(time.RFC3339Nano), msg.ExternalOwner, msg.Namespace, msg.Owner.String(), msg.Pubkey)
+	// Create the StdSignDoc by using the given message data, with an empty string
+	signedMsg := msg
+	signedMsg.Signature = ""
+
+	stdSignDoc := auth.StdSignDoc{
```
Comment (kwunyeung) [inline x/magpie/internal/keeper/handler.go]: Looks good. Would like to know if test pass.

File: x/posts/client/cli/query.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/kwunyeung/desmos/x/posts/internal/types"
+	"github.com/spf13/cobra"
+)
+
+// GetQueryCmd adds the query commands
+func GetQueryCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	magpieQueryCmd := &cobra.Command{
```
Comment (kwunyeung) [inline x/posts/client/cli/query.go]: May use a proper variable name in `Post` module.

File: x/posts/client/cli/tx.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/kwunyeung/desmos/x/posts/internal/types"
+)
+
+// GetTxCmd set the tx commands
+func GetTxCmd(_ string, cdc *codec.Codec) *cobra.Command {
+	magpieTxCmd := &cobra.Command{
```
Comment (kwunyeung) [inline x/posts/client/cli/tx.go]: Same for variable name.

File: x/magpie/internal/types/msgs_test.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+var testOwner, _ = sdk.AccAddressFromBech32("cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns")
+var timeZone, _ = time.LoadLocation("UTC")
+var msgShareDocumentSchema = types.MsgCreateSession{
+	Owner:         testOwner,
+	Created:       time.Date(2019, 10, 31, 9, 42, 0, 0, timeZone),
+	Namespace:     "cosmos",
+	ExternalOwner: "cosmos1njrqah832yfdv8yhxnrskerzxhj5zj9e563uge",
+	PubKey:        "cosmospub1addwnpepqf06gxm8tf4u9af99zsuphr2jmqvr2t956me5rcx9kywmrtg6jewy8gjtcs",
+	Signature:     "QmZh...===",
```
Comment (kwunyeung) [inline x/magpie/internal/types/msgs_test.go]: Does this work?

**Comment by kwunyeung** [comment]
`go mode vendor`

**Comment by RiccardoM** [comment]
> will test soon

It's still WIP. I want to add some more tests and later fix the issues before tagging it R4R. I'll request your review when done. 

> `go mod vendor`

I avoid using `vendor` and prefer `download` instead. Will fix the problems soon btw

**Comment by RiccardoM** [comment]
Closing this as it's getting very dirty

**Pull request closed without merging**